src/main/java/org/springframework/integration/aws/outbound/KplMessageHandler.java

```diff
@@ -115,9 +122,26 @@ public void setConverter(Converter<Object, byte[]> converter) {
 		setMessageConverter(new ConvertingFromMessageConverter(converter));
 	}
 
+	/**
+	 * Configure maximum records in flight for handling backpressure. By Default, backpressure handling is not enabled.
+	 * On number of records in flight exceeding the threshold, {@link KplBackpressureException} would be thrown.
+	 * If Backpressure handling is enabled, {@link KplBackpressureException} must be handled.
+	 *
+	 * @param backPressureThreshold Defaulted to 0. Set a value greater than 0 to enable backpressure handling.
+	 *
+	 * @since 3.0.9
+	 */
+	public void setBackPressureThreshold(long backPressureThreshold) {
+		Assert.isTrue(backPressureThreshold >= 0,
+				"'backPressureThreshold must be greater than equal to 0.");
```

Comment:
What if I would like to disable it?
Looks our logic allows us to use `0`, and this property might be set by some conditional config where value could be resolved to `0`.

Reply:
0 being allowed value, the setter should allow it too. There could be cases, where a user would want to disable after enabling it. Even if 0 is allowed, One wouldn't be able to disable it back, if we have only ```Assert`greater than 0```. Hence I had relaxed it back. Negative value wouldnt mean anything, which is still covered. 

One simple occurrence was in test cases failure in the same suite, the state for the threshold was persisted due to other test cases running in random order.

Reply:
Sorry, missed a `>=` operator.
So, all good here, unless we would fix the exception message with an `or` between `than` and `equal`.